### ISSUE TITLE
Overloaded method with higher kinded type shadows value type on 2.13

### ISSUE THREAD

**Issue by lJoublanc** [scala/bug#11487]
@milessabin @mpilquist has kindly provided this minimized example:

```scala
scala> object Foo {
     |   def apply[A]: Int = 0
     |   def apply[F[_]](implicit x: Int): Boolean = true
     | }
defined object Foo

scala> Foo[Int]
           ^
       error: Int takes no type parameters, expected: one
```
The whole discussion is here :point_up: [April 15, 2019 12:17 PM](https://gitter.im/mpilquist/simulacrum?at=5cb4686125686a7dc3347600). 

Confirmed on both 2.13-M5 and RC1.

This doesn't occur if the methods have a different number of type parameters.

**Comment by hrhino** [comment]
This also happens on 2.12, it seems, so not a 2.13 regression: https://scalafiddle.io/sf/D35fZL1/0

**Comment by joroKr21** [comment]
I wouldn't say it's fixed in dotty. It just chooses the other overload:
```scala
scala> Foo[List]
1 |Foo[List]
  |    ^^^^
  |    Missing type parameter for List
```

**Comment by joroKr21** [comment]
Fixed by scala/scala#9405 but let's wait and see what dotty has to say: lampepfl/dotty#11015

**Comment by joroKr21** [comment]
Confirmed that it's a bug in dotty and therefore the fix that was merged in scalac is the right thing to do.
